From: yangsong8 <yangsong8@xiaomi.com>
Date: Tue, 12 Nov 2024 10:07:57 +0800
Subject: [PATCH] cdcacm: check present bit

When close the serial tool in PC, only the persent bit is cleared
on Windows but all bits are cleared on linux.

Signed-off-by: yangsong8 <yangsong8@xiaomi.com>

diff --git a/drivers/usbdev/cdcacm.c b/drivers/usbdev/cdcacm.c
index 691af901b2..45862ead98 100644
--- a/drivers/usbdev/cdcacm.c
+++ b/drivers/usbdev/cdcacm.c
@@ -3138,7 +3138,7 @@ ssize_t cdcacm_write(FAR const char *buffer, size_t buflen)
     {
       irqstate_t flags;
 
-      if (!priv || !priv->ctrlline)
+      if (!priv || !(priv->ctrlline & CDC_DTE_PRESENT))
         {
           return -EINVAL;
         }
